js/checkRanges.js

```diff
@@ -9,6 +9,24 @@ $('document').ready(function() {
  * If the user enters a value not found in the list of additional valid
  * options, the original REDCap validation is called.
  */
+
+    function isValueOnRange(val) {
+      var ranges = complex_field_validation_tag_values;
```

Comment:
this line needs to receive an array of complex_field_validation_tag_values indexed by field_name. You'll need to add an outer loop to process the complex_field_validation_tag_values for each field_name present in the array.